.github/workflows/build-publish-image.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and Publish Bandit Images
+
+on:
+  release:
+    types: [created]
+  schedule:
+    - cron: '0 0 * * 0' # Every Sunday at midnight
+  workflow_dispatch:
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+
+    steps:
+
+    - name: Get latest release tag
+      if: github.event_name != 'release'
+      id: get-latest-tag
+      run: |
+        TAG=$(curl -s https://api.github.com/repos/${{ github.repository }}/releases/latest | jq -r .tag_name)
+        echo "Latest tag is $TAG"
+        echo "RELEASE_TAG=$TAG" >> $GITHUB_ENV
```

Comment:
I've never seen this construct before, is it possible that `GITHUB_ENV` does not have a terminating `\n` in it? Is that potentially something someone could exploit to publish unreleased code?

Reply:
It's a pretty standard approach for setting key values in actions:

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-writing-an-environment-variable-to-github_env

https://github.com/lukehinds/testrepo/actions/runs/7612691776/job/20730842060
https://github.com/lukehinds/testrepo/blob/main/.github/workflows/test-termination.yml